src/hydro/srcterms/tabular_cooling.cpp

```diff
@@ -751,9 +634,11 @@ Real TabularCooling::EstimateTimeStep(MeshData<Real> *md) const {
                  log_temp_final, d_log_temp, n_temp, log_lambdas, dedt_valid);
 
         // Compute cooling time
-        // If de_dt is zero, using infinite cooling time
-        const Real cooling_time = (de_dt == 0 ? std::numeric_limits<Real>::infinity()
-                                              : fabs(internal_e / de_dt));
+        // If de_dt is zero (temperature is smaller than lower end of cooling table) or
+        // current temp is below floor, use infinite cooling time
+        const Real cooling_time = ((de_dt == 0) || (internal_e < internal_e_floor))
+                                      ? std::numeric_limits<Real>::infinity()
+                                      : fabs(internal_e / de_dt);
```

Comment:
Very good catch. I haven't tested much with a cooling "CFL" when gas is below the cooling table floor.